etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/RemoveMessageTypeStructure.java

```diff
@@ -0,0 +1,38 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.custom;
+
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.FhirResource;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.RuleExecutionException;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.CustomFhirTransformation;
+import gov.hhs.cdc.trustedintermediary.external.hapi.HapiHelper;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Coding;
+
+/** Removes the Message Structure (MSH-9.3) from the Message Type (MSH-9). */
+public class RemoveMessageTypeStructure implements CustomFhirTransformation {
+
+    @Override
+    public void transform(FhirResource<?> resource, Map<String, String> args)
+            throws RuleExecutionException {
+        try {
+            Bundle bundle = (Bundle) resource.getUnderlyingResource();
+            Coding coding = HapiHelper.getMessageTypeCoding(bundle);
+            String display = coding.getDisplay();
+            String delimiter = "^";
+            ArrayList<String> displayList =
+                    Arrays.stream(display.split("\\" + delimiter))
+                            .collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Even better, thanks! I'll change it

Reply:
Very weird.  I would have thought that this would have returned a modifiable list (as its JavaDocs state), but the unit test fails if we don't wrap it with `new ArrayList<>(...)`.

Reply:
Correct